File: tests/codegen/snapshots/python/union_with_typename_and_fragment.py
from typing_extensions import TypedDict
from typing import Optional, Union

class AnimalProjection(TypedDict):
    # typename: Animal
    age: int

class OperationNameResultUnionPerson(TypedDict):
    # typename: Person
    name: str

OperationNameResultUnion = Union[AnimalProjection, OperationNameResultUnionPerson]

class OperationNameResultOptionalUnionPerson(TypedDict):
    # typename: Person
    name: str

OperationNameResultOptionalUnion = Union[AnimalProjection, OperationNameResultOptionalUnionPerson]

class OperationNameResult(TypedDict):
    union: OperationNameResultUnion
    optional_union: Optional[OperationNameResultOptionalUnion]

File: tests/codegen/snapshots/typescript/union_with_typename_and_fragment.ts
type AnimalProjection = {
    age: number
}

type OperationNameResultUnionPerson = {
    name: string
}

type OperationNameResultUnion = AnimalProjection | OperationNameResultUnionPerson

type OperationNameResultOptionalUnionPerson = {
    name: string
}

type OperationNameResultOptionalUnion = AnimalProjection | OperationNameResultOptionalUnionPerson

type OperationNameResult = {
    __typename: string
    union: OperationNameResultUnion
    optional_union: OperationNameResultOptionalUnion | undefined
}
